pyspider/fetcher/puppeteer_fetcher.js

```diff
@@ -0,0 +1,275 @@
+const express = require("express");
+const puppeteer = require('puppeteer');
+const bodyParser = require('body-parser');
+
+const app = express();
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+    extended: false
+}));
+
+async function spider(fetch, http_response) {
+    var first_request = true,
+        first_response = null,
+        finished = false,
+        page_loaded = false,
+        start_time = Date.now(),
+        end_time = null,
+        script_executed = false,
+        script_result = null,
+        error_msg = null,
+        wait_before_end = 1000,
+        browser_settings = {};
+
+    if (fetch.headless == "false") {
+        browser_settings["headless"] = false;
+    } else {
+        browser_settings["headless"] = true;
+    }
+
+    if (fetch.proxy) {
+        if (fetch.proxy.indexOf("://") == -1) {
+            fetch.proxy = "http://" + fetch.proxy;
+        }
+        browser_settings["args"] = ['--no-sandbox', "--proxy-server=" + fetch.proxy];
+    }
+
+    const browser = await puppeteer.launch(browser_settings);
```

Comment:
It will have performance issue if new browser launched for each request. Browser and tabs should be pooled and reused.